internal/service/cluster/data_source_clusters.go

```diff
@@ -384,8 +388,14 @@ func flattenClusters(ctx context.Context, d *schema.ResourceData, conn *matlas.C
 			log.Printf("[WARN] Error setting `snapshot_backup_policy` for the cluster(%s): %s", clusters[i].ID, err)
 		}
 
-		processArgs, _, err := conn.Clusters.GetProcessArgs(ctx, clusters[i].GroupID, clusters[i].Name)
-		log.Printf("[WARN] Error setting `advanced_configuration` for the cluster(%s): %s", clusters[i].ID, err)
+		processArgs20240530, _, err := connV220240530.ClustersApi.GetClusterAdvancedConfiguration(ctx, clusters[i].GroupID, clusters[i].Name).Execute()
+		if err != nil {
+			log.Printf("[WARN] Error setting `advanced_configuration` for the cluster(%s): %s", clusters[i].ID, err)
+		}
+		processArgs, _, err := connV2.ClustersApi.GetClusterAdvancedConfiguration(ctx, clusters[i].GroupID, clusters[i].Name).Execute()
```

Comment:
why making two calls here? in what cases using the 0530 API version can fail and the other doesn't?

Reply:
0530 version is being called to continue supporting `default_read_concern` and `fail_index_key_too_long` which are deprecated now and no longer available in newer versions of the SDK.